components/ArticleItemWidget/ArticleItemWidget.js

```diff
@@ -0,0 +1,255 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import NProgress from 'nprogress';
+import cx from 'classnames';
+
+import gql from '../../util/gql';
+import { articleItemWidgetStyle } from './ArticleItemWidget.styles';
+
+import NotArticleExtendSelection from './NotArticleExtendSelection.js';
+
+// TODO: Click out for touch
+
+// without state for tags (like: hover, show, hide...)
+const initialState = {
+  barOpen: false,
+  notArticleSelectionDisplay: false,
+  touchTarget: false, // record the target now touch(move!) over by dom id, not touch start.
+};
+
+export default class ArticleItemWidget extends PureComponent {
+  static propTypes = {
+    id: PropTypes.string.isRequired, // Article ID
+    read: PropTypes.bool.isRequired,
+    notArticleReplied: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.string, // TYPE_SUGGESTION_OPTIONS in replyType.js
+    ]),
+    handleLocalEditorHelperList: PropTypes.func.isRequired,
```

Comment:
Let's keep `handleXXX` to the actual "handler" that implements the change.

`ArticleItemWidget` is invoking the callback "on" some event, so we should use `onXXX` to name the handlers here.

I think `onLocalEditorHelperListChange` would be a bit too long, `onChange` would be appropriate for this case since `ArticleItemWidget` could only "change" in one way.